src/UCTNode.cpp

```diff
@@ -412,133 +349,75 @@ UCTNode* UCTNode::uct_select_child(int color) {
 
         if (value > best_value) {
             best_value = value;
-            best = child;
-        }
-
-        child = child->m_nextsibling;
-        // Make sure we are at a valid successor.
-        while (child != nullptr && !child->valid()) {
-            child = child->m_nextsibling;
+            best = child.get();
         }
     }
 
     assert(best != nullptr);
```

Comment:
The old code allowed to return `nullptr` when all children were invalid.
In debug mode we would now assert.

Reply:
IMHO best can never be nullptr, there will always be one valid child (pass).

as a smoke test I've also tested a couple of games in debug mode without any asserts being thrown

Reply:
With a 2nd look I'm confident this behavior hasn't changed, the assert was there before. the loop accesses evaluates the same children